src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,139 @@
+package task;
+
+import exception.TaskParseException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Implementation of a <code>Task</code> with one chronological deadline.
+ */
+public class Deadline extends Task {
+    private final LocalDateTime by;
+
+    /**
+     * Creates a deadline with the specified objective,
+     * and the specified deadline timing.
+     *
+     * @param objective description of this deadline's objective.
+     * @param by        deadline time of this deadline.
+     */
+    public Deadline(String objective, LocalDateTime by) {
+        super(objective);
+        this.by = by;
+    }
+
+    /**
+     * Parses the supplied <code>String[]</code> command-line arguments to create a deadline.
+     *
+     * @param args arguments containing the deadline to be parsed.
+     * @return a deadline represented by <code>args</code>.
+     * @throws TaskParseException if <code>args</code> does not represent a valid deadline.
+     */
+    public static Deadline parseArgs(String[] args) throws TaskParseException {
+        String objective = "";
+        String by = "";
+        boolean isInTokenBy = false;
```

Comment:
Perharps can explain what is the situation that this boolean is used for.

Reply:
Added a little comment to that line & similar lines elsewhere 👍